elfpy/types.py

```diff
@@ -126,9 +126,12 @@ def __neg__(self):
         return Quantity(amount=-self.amount, unit=self.unit)
 
 
+T = TypeVar("T")
+
+
 @dataclass
-class Trade:
+class Trade(Generic[T]):
     r"""A trade for a market"""
 
     market: MarketType
-    trade: Any  # TODO: How to specify the type as a generic market action?
+    trade: T  # TODO: How to specify the type as a generic market action?
```

Comment:
we have a MarketActionType in base_market.py, but the MarketActionTypes in borrow_market.py and hyperdrive_actions.py don't inherit from it. so we'd need a new parent class?